src/libcore/cell.rs

```diff
@@ -521,6 +545,20 @@ impl<T: Default> Cell<T> {
 #[unstable(feature = "coerce_unsized", issue = "27732")]
 impl<T: CoerceUnsized<U>, U> CoerceUnsized<Cell<U>> for Cell<T> {}
 
+#[unstable(feature = "as_cell", issue="43038")]
+impl<T, I> Index<I> for Cell<[T]>
+where
+    I: SliceIndex<[Cell<T>]>
+{
```

Comment:
Oh, something potentially of note: it should be possible to implement `Deref<Target=[Cell<T>]>` on `Cell<[T]>` (instead of `Index`), to allow e.g. calling `.len()` directly.